components/switch/switch-mixin.js

```diff
@@ -0,0 +1,181 @@
+
+	static get properties() {
+		return {
+			disabled: { type: Boolean, reflect: true },
+			label: { type: String, reflect: true },
+			labelHidden: { type: Boolean, attribute: 'label-hidden', reflect: true },
+			on: { type: Boolean, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: inline-block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+
+			.d2l-switch-container {
+				background-color: #ffffff;
+				border: 2px solid transparent;
+				border-radius: 1rem;
+				box-sizing: border-box;
+				cursor: pointer;
+				display: inline-block;
+				font-size: 0;
+				line-height: 0;
+				outline-style: none;
+				padding: 0.1rem;
+				vertical-align: middle;
+			}
+			.d2l-switch-container:focus,
+			.d2l-switch-container:hover {
+				border-color: var(--d2l-color-celestine);
+			}
+			:host([disabled]) .d2l-switch-container {
+				cursor: default;
+				opacity: 0.5;
+			}
+			:host([disabled]) .d2l-switch-container:focus,
+			:host([disabled]) .d2l-switch-container:hover {
+				border-color: transparent;
+			}
+
+			.d2l-switch-inner {
+				border: 1px solid var(--d2l-color-ferrite);
+				border-radius: 0.8rem;
+				box-sizing: border-box;
+				padding: 0.3rem;
+			}
+			:host([on]) .d2l-switch-inner {
+				background-color: var(--d2l-color-celestine-plus-2);
+				border-color: var(--d2l-color-celestine);
+			}
+
+			.d2l-switch-toggle {
+				position: relative;
+				transition: transform 150ms linear;
+			}
+			.d2l-switch-toggle > div {
+				background-color: #ffffff;
+				border: 1px solid var(--d2l-color-ferrite);
+				border-radius: 0.6rem;
+				box-sizing: border-box;
+				display: inline-block;
+				height: 1.2rem;
+				left: -0.2rem;
+				position: absolute;
+				top: -0.2rem;
+				width: 1.2rem;
+			}
+			:host([dir="rtl"]) .d2l-switch-toggle > div {
+				left: auto;
+				right: -0.2rem;
+			}
+			:host([on]) .d2l-switch-toggle {
+				transform: translateX(1.2rem);
+			}
+			:host([dir="rtl"][on]) .d2l-switch-toggle {
+				transform: translateX(-1.2rem);
+			}
+			:host([on]) .d2l-switch-toggle > div {
+				border-color: var(--d2l-color-celestine);
+			}
+
+			d2l-icon, d2l-icon-custom {
+				height: 0.8rem;
+				width: 0.8rem;
+			}
+			.d2l-switch-icon-on, .d2l-switch-icon-off {
+				display: inline-block;
+			}
+			.d2l-switch-icon-on {
+				margin-right: 0.4rem;
+			}
+			:host([dir="rtl"]) .d2l-switch-icon-on {
+				margin-left: 0.4rem;
+				margin-right: 0;
+			}
+			:host([on]) .d2l-switch-icon-on > d2l-icon,
+			:host([on]) .d2l-switch-icon-on > d2l-icon-custom {
+				color: var(--d2l-color-celestine);
+			}
+
+			.d2l-switch-label {
+				font-size: 0.8rem;
+				font-weight: 400;
+			}
+
+			@media (prefers-reduced-motion: reduce) {
+				.d2l-switch-toggle {
+					transition: none;
+				}
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this.disabled = false;
+		this.labelHidden = false;
+		this.on = false;
+		this._labelId = getUniqueId();
+	}
+
```

Comment:
I'm assuming a label is required? If so, worth `console.warn`-ing in `firstUpdated` if they forget to provide one? I'd like to start using warnings more often for invalid usages like this. We could even get really aggressive and throw from `render()`...

Reply:
Cool yep... add a warning.  A label is indeed required.